### PR TITLE
chore: bump operator-rs to 0.60.1

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

- Bump operator-rs to 0.60.1
- **BREAKING** Removed legacy node selector on roleGroups.
- Use new label builders.
- Use Snafu instead of thiserror (part of: https://github.com/stackabletech/operator-rs/issues/694)

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by NickLarsenNZ** [comment]
Running kuttl tests in Azure

File: rust/operator-binary/src/hbase_controller.rs [hunk 1]
```diff
@@ -226,6 +257,12 @@ pub enum Error {
     GracefulShutdown {
         source: crate::operations::graceful_shutdown::Error,
     },
+
+    #[snafu(display("failed to build label"))]
+    BuildLabel { source: LabelError },
+
+    #[snafu(display("failed to build object  meta data"))]
```
Comment (sbernauer) [inline rust/operator-binary/src/hbase_controller.rs:264]:
```suggestion
    #[snafu(display("failed to build object meta data"))]
```

**Review by sbernauer — APPROVED** [review]
LGTM otherwise

File: rust/operator-binary/src/hbase_controller.rs (hunk 1 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/hbase_controller.rs:264]: Oof, copypasta from https://github.com/stackabletech/zookeeper-operator/blob/bb9d6fa638b02d743cdf8fbf58a729048d90cb21/rust/operator-binary/src/zk_controller.rs#L240-L241

Comment (NickLarsenNZ) [reply rust/operator-binary/src/hbase_controller.rs]: Fixed in 45025e2

**Review by sbernauer — APPROVED** [review]

**Pull request merged**